olive/resource_path.py

```diff
@@ -0,0 +1,387 @@
+        path = Path(v)
+        if not path.is_dir():
+            raise ValueError(f"Path {path} is not a folder.")
+        return path
+
+
+class StringNameConfig(ResourceConfig):
+    """Config for a string name."""
+
+    _type = ResourceType.StringName
+    name: str = Field(..., description="Name of the resource.")
+
+    def get_path(self) -> str:
+        return self.name
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        logger.debug("Resource is a string name. No need to save to directory.")
+        return self.name
+
+
+def _get_azureml_resource_prefix(workspace_config: Dict[str, str]) -> str:
+    return (
+        f"azureml://subscriptions/{workspace_config['subscription_id']}"
+        f"/resourcegroups/{workspace_config['resource_group']}"
+        f"/workspaces/{workspace_config['workspace_name']}"
+    )
+
+
+class AzureMLModelConfig(ResourceConfig):
+    """Config for an AzureML model."""
+
+    _type = ResourceType.AzureMLModel
+    azureml_client: AzureMLClientConfig = Field(..., description="AzureML client config.")
+    name: str = Field(..., description="Name of the model.")
+    version: Union[int, str] = Field(..., description="Version of the model.")
+
+    def get_path(self) -> str:
+        return f"azureml:{self.name}:{self.version}"
+
+    def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: bool = False) -> str:
+        # directory to save the resource to
+        dir_path = Path(dir_path).resolve()
+        dir_path.mkdir(parents=True, exist_ok=True)
+
+        # azureml client
+        ml_client = self.azureml_client.create_client()
+
+        # azureml model
+        model = ml_client.models.get(self.name, self.version)
+        model_path = Path(model.path)
+
+        # path to save the resource to
+        if name:
+            new_path_name = Path(name).with_suffix(model_path.suffix).name
+        else:
+            new_path_name = model_path.name
+        new_path = dir_path / new_path_name
+        _overwrite_helper(new_path, overwrite)
+
+        # download the resource to the new path
+        try:
+            logger.debug(f"Downloading model {self.name} version {self.version} to {new_path}.")
+            from azure.core.exceptions import ServiceResponseError
+
+            with tempfile.TemporaryDirectory(dir=dir_path, prefix="olive_tmp") as temp_dir:
+                temp_dir = Path(temp_dir)
+                retry_func(
+                    ml_client.models.download,
+                    [self.name],
+                    {"version": self.version, "download_path": temp_dir},
+                    max_tries=3,
+                    delay=5,
+                    exceptions=ServiceResponseError,
+                )
+                new_path.parent.mkdir(parents=True, exist_ok=True)
+                shutil.move(temp_dir / self.name / model_path.name, new_path)
+        except Exception as e:
```

Comment:
the try/catch logging is needless since the caller will catch the exception and handle it

Reply:
I added the try except for logging the error but I can remove it too. Do you think the error logging is not needed?

Reply:
If the exception is not rethrown, the catch and log is fine. I prefer remove it because the caller will catch the rethown exception and do error handling. From this point of view, it seems the logging is duplicated?

Reply:
Done. I removed the try/except part since the logging is redundant as you mentioned.